src/locale-proxy.ts

```diff
@@ -0,0 +1,98 @@
+import type { LocaleDefinition } from './definitions';
+import { FakerError } from './errors/faker-error';
+
+/**
+ * A proxy for LocaleDefinitions that marks all properties as required and throws an error when an entry is accessed that is not defined.
+ */
+export type LocaleAccess = Readonly<{
+  [key in keyof LocaleDefinition]-?: Readonly<
+    Required<NonNullable<LocaleDefinition[key]>>
+  >;
+}>;
+
+/**
+ * Creates a proxy for LocaleDefinition that throws an error if an undefined property is accessed.
+ *
+ * @param locale The locale definition to create the proxy for.
+ */
+export function createLocaleAccess(locale: LocaleDefinition): LocaleAccess {
+  return new Proxy({} as LocaleDefinition, {
+    has(): true {
+      return true;
+    },
+
+    get(
+      target,
+      categoryName: keyof LocaleDefinition
+    ): LocaleDefinition[keyof LocaleDefinition] {
+      if (categoryName in target) {
+        return target[categoryName];
+      }
+
+      return (target[categoryName] = createCategoryProxy(
+        categoryName,
+        locale[categoryName]
+      ));
+    },
+
+    set(): never {
+      throw new FakerError('LocaleAccess is read-only.');
```

Comment:
I'm not sure if this error makes sense to the user since they never directly interact with `LocalAccess`. Maybe something more generic like:

```suggestion
      throw new FakerError('Locale data set's are read-only.');
```